readchar/readchar.py

```diff
@@ -1,30 +1,26 @@
 # -*- coding: utf-8 -*-
-# This file is based on this gist:
-# http://code.activestate.com/recipes/134892/
-# So real authors are DannyYoo and company.
+# Copyright (c) 2014- 2017 Miguel Ángel García (@magmax9).
+# Based on previous work on gist getch()-like unbuffered character
+# reading from stdin on both Windows and Unix (Python recipe),
+# started by Danny Yoo. Licensed under the MIT license.
+
+from __future__ import absolute_import
 import sys
 
 
-if sys.platform.startswith('linux'):
-    from .readchar_linux import readchar
-elif sys.platform == 'darwin':
-    from .readchar_linux import readchar
-elif sys.platform in ('win32', 'cygwin'):
-    from .readchar_windows import readchar
+platform = sys.platform[0:5]
+if platform in ("linux", "darwi"):
+    from .readchar_linux import get_char, read_char
+elif platform in ('win32', 'cygwin'):
+    from .readchar_windows import get_char, read_char
 else:
     raise NotImplemented('The platform %s is not supported yet' % sys.platform)
 
 
-def readkey(getchar_fn=None):
-    getchar = getchar_fn or readchar
-    c1 = getchar()
-    if ord(c1) != 0x1b:
-        return c1
-    c2 = getchar()
-    if ord(c2) != 0x5b:
-        return c1 + c2
-    c3 = getchar()
-    if ord(c3) != 0x33:
-        return c1 + c2 + c3
-    c4 = getchar()
-    return c1 + c2 + c3 + c4
+def readkey(test_stream=None):
+    charbuffer = get_char(test_stream)
+    return charbuffer
+
+
+def readchar(blocking=True):
+    return read_char(blocking)
```

Comment:
I made the interface with a default so that the old way works the same as before (blocks waiting for input from keyboard).  Both readkey and readchar "wait" (Block) for a user to press the key.  Readchar and readkey are already documented in your docs.. 

```
readchar()

Reads the next char from stdin, returning it as a string with length 1.

readkey()

Reads the next key-stroke from stdin, returning it as a string.
```

So readkey could be a single string like "[[^1" (or something like that), whereas, readchar would return the same keypress as four characters using four different events: '[', followed by '[', '^', and '1'

both of them "wait" (block) and do not return until a user presses the key...

Another example (and maybe the difference) would be if two keys were pressed, readkey would return two things: "[[^1" and "[[^2".  readchar would return eight: '[', '[', '^', '1', '[', '[', '^', and '2'